scrapy/spider.py

```diff
@@ -31,11 +31,19 @@ def __init__(self, name=None, **kwargs):
         if not hasattr(self, 'start_urls'):
             self.start_urls = []
 
-    def log(self, message, level=log.DEBUG, **kw):
-        """Log the given messages at the given log level. Always use this
-        method to send log messages from your spider
+    @property
+    def logger(self):
+        logger = logging.getLogger(self.name)
```

Comment:
>how log messages will be propagated if logger is not a child of 'scrapy' logger?

This messages aren't propagated on purpose, since they are issued by the user and not Scrapy. Since this is actually a helper for creating a logger and logging through it directly, and we can't control what loggers the users create, I think it makes more sense to detach it from Scrapy logging.